src/main/java/org/wso2/lsp4intellij/utils/FileUtils.java

```diff
@@ -232,13 +234,37 @@ public static String sanitizeURI(String uri) {
                     if (!reconstructed.toString().endsWith(":")) {
                         reconstructed.append(":");
                     }
-                    return reconstructed.append(uriCp.substring(uriCp.indexOf(URI_PATH_SEP))).toString();
+                    String result = reconstructed.append(uriCp.substring(uriCp.indexOf(URI_PATH_SEP))).toString();
+                    return lowercaseWindowsDriveAndEscapeColon(result);
+
                 }
             }
         } else {
             return null;
         }
     }
+   /**
+     * The LSP specification <a href="https://microsoft.github.io/language-server-protocol/specification/#uri">requires</a>
+     * all servers to handle two URI formats correctly: {@code file:///C:/foo} and {@code file:///c%3A/foo}.
+     * <p>
+     * VS Code always sends lowercase Windows drive letters and always escapes colons
+     * (see this <a href="https://github.com/microsoft/vscode-languageserver-node/issues/1280">issue</a>
+     * and related <a href="https://github.com/microsoft/language-server-protocol/pull/1786">pull request</a>).
+     * <p>
+     * Some LSP servers only support VS Code friendly URI format ({@code file:///c%3A/foo}),
+     * so it's safer to use this format by default.
+     */
+    public static String lowercaseWindowsDriveAndEscapeColon(String uri) {
```

Comment:
We (1c-syntax) wrote Absolute class to handle uris in the same way in all places: 
https://github.com/1c-syntax/utils/blob/develop/src/main/java/com/github/_1c_syntax/utils/Absolute.java